src/Lunr/Vortex/PushNotificationDispatcher.php

```diff
@@ -94,9 +103,17 @@ public function dispatch(array $endpoints, array $payloads): void
 
             if (!isset($this->dispatchers[$platform]))
             {
-                foreach ($grouped_endpoints[$platform] as $payload_endpoints)
+                if (isset($grouped_endpoints[$platform]))
                 {
-                    $this->statuses[Status::NotHandled->value] = array_merge($this->statuses[Status::NotHandled->value], $payload_endpoints);
+                    foreach ($grouped_endpoints[$platform] as $payload_endpoints)
+                    {
+                        $this->statuses[Status::NotHandled->value] = array_merge($this->statuses[Status::NotHandled->value], $payload_endpoints);
+                    }
+                }
+
+                foreach (array_filter($platform_payloads, fn($payload) => $payload->is_broadcast()) as $payload)
+                {
+                    $this->broadcast_statuses[Status::NotHandled->value][] = $payload;
```

Comment:
Conceptually, we can have more than one broadcast per dispatch (One for FCM, one for APNS, etc). `$this->statuses` is indexed, so we can retrieve the status for a single endpoint. `$this->broadcast` should be indexed as well, so we can retrieve the status for a specific broadcast.

Reply:
I indexed the payloads by platform and payload type